src/components/ResourceInstance/Logs/SyntaxHighlightedLog.tsx

```
@@ -0,0 +1,387 @@
+  if (!enableSyntaxHighlighting) {
+    return (
+      <HighlightedLogContent logType={null}>
+        <Ansi>{logLine}</Ansi>
+      </HighlightedLogContent>
+    );
+  }
+
+  // Apply highlighting if we detected a format
+  if (detectedFormat) {
+    const highlightedText = applyBasicHighlighting(logLine, detectedFormat);
+    return <HighlightedLogContent logType={detectedFormat} dangerouslySetInnerHTML={{ __html: highlightedText }} />;
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)